neurons/validator.py

```diff
@@ -370,26 +376,57 @@ def update_models(self):
                         current_uid_count = len(self.uids_to_eval)
 
                 # Get the next uid to check
-                next_uid = next(self.miner_iterator)
-
-                # Confirm that we haven't checked it in the last 5 minutes.
-                time_diff = (
-                    dt.datetime.now() - uid_last_checked[next_uid]
-                    if next_uid in uid_last_checked
-                    else None
-                )
+                next_uid = None
 
-                if time_diff and time_diff < dt.timedelta(minutes=5):
-                    # If we have seen it within 5 minutes then sleep until it has been at least 5 minutes.
-                    time_to_sleep = (
-                        dt.timedelta(minutes=5) - time_diff
-                    ).total_seconds()
-                    bt.logging.trace(
-                        f"Update loop has already processed all UIDs in the last 5 minutes. Sleeping {time_to_sleep} seconds."
+                # First check for any uids with incentives above threshold on the chain.
+                # This will catch updates to current best models and models other valis have incentivized faster.
+                incentives = None
+                with self.metagraph_lock:
+                    incentives = self.metagraph.I
+
+                for uid, incentive in enumerate(incentives):
+                    # Use .item() to get the number value since this is a tensor.
+                    if (
+                        incentive.item()
+                        >= constants.update_priority_incentive_threshold
+                    ):
+                        # Confirm that we haven't checked it in the last 30 minutes in this path.
```

Comment:
Should we make this 20 minutes to align with the cadence the chain metadata can be updated?

Reply:
Done. And below.